### PR TITLE
Update of running RemRem-Publish documentation.

### PR DESCRIPTION
**Pull request by tobiasake**
Update of running RemRem-Publish documentation.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java [hunk 1]
```diff
@@ -15,9 +15,16 @@
 package com.ericsson.eiffel.remrem.publish.config;
 
 import com.ericsson.eiffel.remrem.publish.constants.RemremPublishServiceConstants;
+
+import ch.qos.logback.classic.Logger;
```
Comment (SantoshNC68) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java]: unused import ?

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java [hunk 2]
```diff
@@ -15,9 +15,16 @@
...
 import io.swagger.annotations.Api;
+
+import org.slf4j.LoggerFactory;
```
Comment (SantoshNC68) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java]: unused import?

**Comment by tobiasake** [comment]
Updated documentation about how to run publish-service and publish-cli with mvn spring-boot:run.
The only issue with mvn spring-boot:run is that it misses the RemRem version in manifest and due to the "/version" entrypoint fails with this exception: 2019-08-29 14:55:49.681 ERROR 7029 --- [nio-8080-exec-2] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.StringIndexOutOfBoundsException: String index out of range: -1] with root cause

java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1967) ~[na:1.8.0_201]
	at com.ericsson.eiffel.remrem.shared.VersionService.getServiceVersion(VersionService.java:114) ~[eiffel-remrem-shared-2.0.1.jar:na]

Eiffel-Shared library tries to find remrem-version in manifest file which is included/added in war file, but is not added in target classes directories when mvn spring-boot:run is executed. But /version entrypoint is not used that often from a development perspective, so we maybe can live with this issue.

File: wiki/markdown/running-remrem-publish.md [hunk 3]
```diff
@@ -4,18 +4,28 @@ RemRem-Publish can be run from source code with maven command and with RemRem-Pu
 
 ## Maven
 
+### Running RemRem-Publish service
+
 To run RemRem-Publish from source code root directory with maven command and RemRem-Publish Spring/Java properties provided on command line:
 
-    mvn spring-boot:run -Dlogging.level.root=DEBUG -Dgenerate.server.uri=http://127.0.0.1:8080
+    mvn -pl publish-service -am spring-boot:run -Dlogging.level.root=DEBUG -Dgenerate.server.uri=http://127.0.0.1:8080
```
Comment (SantoshNC68) [inline wiki/markdown/running-remrem-publish.md]: can we also mention how to run it from the service itself without having to go for root directory?

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java (hunk 1 above)
Comment (tobiasake) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java]: fixed.

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java (hunk 2 above)
Comment (tobiasake) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/config/SwaggerConfig.java]: fixed.

File: wiki/markdown/running-remrem-publish.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/running-remrem-publish.md]: How do you mean?  You have example how to execute with with Java command and in Tomcat further down on this page.
mvn spring-boot:run works only from root directory, due to how pom files structure is in RemRem-Publish and all version properties/variables in defined in root dir pom file.

**Comment by tobiasake** [comment]
Can't get the Swagger page RemRem API version to work in both situatuions how RemRem-Publish is executed, mvn spring-boot:run and java -jar publish-service.war. When mvn spring-boot:run manifest is not generated and version can't be read from manifest. So changing this pull request to only be about update of how to run RemRem-Publish documetntation.

File: wiki/markdown/running-remrem-publish.md (hunk 3 above)
Comment (SantoshNC68) [reply wiki/markdown/running-remrem-publish.md]: ok, if it doesn't run from the individual projects then i guess this looks good.

Comment (tobiasake) [reply wiki/markdown/running-remrem-publish.md]: If you replace the variables with real version numbers, not variables, in e.g. publish-service/pom.xml,, then it works to execute mvn spring-boot:run in the publish-service folder. So its only possible to execute from root dir as described above.

**Pull request merged**